pkg/maps/policymap/policymap.go

```diff
@@ -44,9 +49,12 @@ const (
 	AllPorts = uint16(0)
 )
 
+var log = logging.DefaultLogger
+
 type PolicyMap struct {
-	path string
-	Fd   int
+	path  string
+	Fd    int
+	mutex lock.Mutex
```

Comment:
As a general style thing, are we deliberately avoiding the following style?

```
type PolicyMap struct {
    lock.Mutex
    ...
}
```

I kinda like the form of `policymap.Lock()`.

(Just starting a discussion, don't bother addressing this in this PR - we can follow up separately)

Reply:
While I sneak the embedded lock style into my PRs (no one ever says anything!), one reason to keep with the in-place style is that our code doesn't keep lock usage inside objects. We often (almost always?) lock an object outside its methods, and modify state. This leads to more public variables on these objects, and that means there is no easy way to distinguish between fields that will change and ones that won't (and so don't need the lock and can be exposed directly). In some cases we place the named `mutex lock.Mutex` above the fields it protects to make it more clear what the boundaries are. I don't think we're terribly consistent, though.